src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -237,6 +240,106 @@ internal async Task<IIdxResponse> IntrospectAsync(IIdxContext idxContext, Cancel
                 request, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <inheritdoc/>
+        public async Task<WidgetSignInResponse> StartWidgetSignInAsync(CancellationToken cancellationToken = default)
```

Comment:
Is the team agreement to move this logic to the client?

Reply:
I'll move it for now.

Reply:
After trying to move it I remember why I put it in the client.  The calling application needs access to the IdxContext that results from a call to InteractAsync(...) so that information therein can be used to initialize (and potentially re-initialize) the widget.  Since the InteractAsync(...) method is internal, the only way to accomplish this, that I could see, was to add `StartWidgetSignInAsync` to the client.

Reply:
I believe the decision was left up to the sdk.  Golang does not require the IDX sdk for any SIW work, while .NET does. At this point in time, we opted to just leave things on this direction.  Going forward and for future development, I believe we should create a small library that the IDX sdk can depend on that provides only the methods for `Interaction_Handle` and `PKCE Generation` code.

Reply:
If that is the direction we want to follow in the future then I think it's better to move this out from here to avoid breaking changes in the future and tell devs "you now need a separate library to implement the self-hosted widget". What do you guys think?

Reply:
My preference, purely for simplicity, would be to keep it here and mark it deprecated later.  But I'm open to whatever we decide.

Reply:
@arvindkrishnakumar-okta Do you have an opinion here ⬆️ ?

Reply:
My vote would be to keep it here for now until we finalize on whether we would be implementing a new library for self-hosted widget APIs.

Reply:
> My vote would be to keep it here for now until we finalize on whether we would be implementing a new library for self-hosted widget APIs or not.

Thanks @arvindkrishnakumar-okta ! @laura-rodriguez fyi ⬆️